site/app/controllers/AuthenticationController.php

```diff
@@ -308,4 +309,200 @@ public function userSelection() {
 
         return new WebResponse(AuthenticationView::class, 'userSelection', $users);
     }
+
+    /**
+     * Display the form for creating a new account
+     */
+    #[Route("/authentication/create_account")]
+    public function signupForm(): ResponseInterface {
+        // Check if the user is already logged in, if yes, redirect to home or another appropriate page
+        if ($this->logged_in) {
+            return new RedirectResponse($this->core->buildUrl(['home']));
+        }
+        return new WebResponse('Authentication', 'signupForm', ['email' => $this->getAcceptedEmails(), 'user_id' => $this->getUserIdRequirements()]);
```

Comment:
I'd recommend adding another check here to see if user registration is enabled before rendering the page.

Reply:
This has been added.